app/javascript/packages/request/index.ts

```diff
@@ -12,10 +12,43 @@ interface RequestOptions extends RequestInit {
   csrf?: boolean | CSRFGetter;
 }
 
-const getCSRFToken = () =>
-  document.querySelector<HTMLMetaElement>('meta[name="csrf-token"]')?.content;
+class CSRF {
```

Comment:
"This" being the Singleton to hold the global values? Seems like a good fit for this case.

One idea could be to have the setters write back to the DOM in case other JS code doesn't use this class? 🤷

Reply:
ah whoops I missed that. perfect!